### PR TITLE
Replaced SocketW with KISSnet

### PR DESCRIPTION
**Pull request by only-a-ptr**
Since we're having no luck stabilizing the rorserver as-is, I'm trying a radical cut: replace dead and suspicious SocketW with https://github.com/Ybalrid/kissnet. I'm hoping to switch to non-blocking sockets to get rid of the threading madness.

UPDATE: It works OK now, the `ERROR Listener: receiving first message` bug was entirely my fault, I had to learn [some](https://www.internalpointers.com/post/c-rvalue-references-and-move-semantics-beginners) [more](https://en.cppreference.com/w/cpp/language/move_constructor) C++.

UPDATE2: Rebased on top of master (after merging #90) and incorporated https://github.com/Ybalrid/kissnet/issues/5. Everything works now. It seems RoR currently crashes on disconnect, but I blame that on SocketW in RoR calling `abort()` again. I'll keep working on it.

### CODE REVIEW

**Comment by tritonas00** [comment]
Compiled, trying to connect RoR:

kk

**Comment by Ybalrid** [comment]
@only-a-ptr You need to upgrade (and configure) your travis (and Linux in general) toolchain to a `C++17` compliant compiler.

https://travis-ci.org/RigsOfRods/ror-server/builds/590908101#L311

I suggest you to add this directive in CMakeLists.txt:

`set(CMAKE_CXX_STANDARD 17)`

Do that *should* be enough under Ubuntu Xenial's GCC version, and turn your Linux build to green (or at least, stop crashing on `kissnet.hpp`)

**Comment by CuriousMike56** [comment]
Is this ready for testing?

**Comment by only-a-ptr** [comment]
@CuriousMike56 Unfortunately no, just yesterday I studied sockets/networking some more and discovered this PR is doomed for 2 reasons:
* The "cleaner" method I used to stop recv. threads is actually windows-only.
* There isn't really any clean method to stop recv. threads on *nix - all information sources I read directly or indirectly suggest that recv. threads are actually "the way it's not supposed to be done".

In short: rorserver is currently made in a way which makes it impossible to disconnect cleanly. Just as I write this I'm looking into http://libevent.org/ and it's C++ derivative https://github.com/Qihoo360/evpp

**Comment by CuriousMike56** [comment]
Ouch..alrighty then. Please work on this then: https://github.com/RigsOfRods/ror-server/pull/97

**Comment by only-a-ptr** [comment]
@CuriousMike56 Sorry, I can't be bothered to add features to a broken piece of software, even if it 'almost' works. I'll keep looking for an actual solution.
